EpiAware/test/EpiObsModels/modifiers/LatentDelay.jl

```diff
@@ -148,16 +148,156 @@ end
         missing_mdl = generate_observations(pois_obs_model, missing, I_t)
-        missing_draws = missing_mdl()
+        missing_draws = missing_mdl() |> Vector{Union{Int, Missing}}
         @test all(ismissing.(missing_draws[1:2]))
         @test !any(ismissing.(missing_draws[3:end]))
 
-        mdl = generate_observations(pois_obs_model, [10.0, 20.0, 30.0, 40.0, 50.0], I_t)
-        @test mdl() == [10.0, 20.0, 30.0, 40.0, 50]
+        mdl = generate_observations(pois_obs_model, [10, 20, 30, 40, 50], I_t)
+        @test mdl() == [10, 20, 30, 40, 50]
         samples = sample(mdl, Prior(), 10; progress = false)
         exp_y_t = get(samples, :exp_y_t).exp_y_t
         @test exp_y_t[1][1] == expected_obs[3]
         @test exp_y_t[2][1] == expected_obs[4]
         @test exp_y_t[3][1] == expected_obs[5]
     end
 end
+
+@testitem "LatentDelay parameter recovery with log-RW latent process: Negative binomial errors" begin
+    using Random, Turing, Distributions, LinearAlgebra, DynamicPPL, StatsBase,
+          ReverseDiff, LogDensityProblems, LogDensityProblemsAD
+    Random.seed!(1234)
+
+    latent_process = RandomWalk(
+        init_prior = Normal(log(100.0), 0.25), std_prior = HalfNormal(0.25))
+    obs_error_model = NegativeBinomialError(cluster_factor_prior = HalfNormal(0.05))
+    d_delay = Gamma(3, 7 / 3)
+    obs_model = LatentDelay(obs_error_model, d_delay)
+
+    @model function test_negbin_errors_with_delays(rw, obs, y_t)
+        n = length(y_t)
+        @submodel Z_t = generate_latent(rw, n)
+        @submodel gen_y_t = generate_observations(obs, y_t, exp.(Z_t))
+        return exp.(Z_t), gen_y_t
+    end
+
+    #Generate data from model
+    generative_mdl = test_negbin_errors_with_delays(
+        latent_process, obs_model, Vector{Union{Missing, Int}}(undef, 50))
+    θ_true = rand(generative_mdl)
+    Z_t_obs, y_t_obs = condition(generative_mdl, θ_true)()
+
+    #Sample from model conditional on data
+    mdl = test_negbin_errors_with_delays(latent_process, obs_model, y_t_obs)
+    ad = AutoReverseDiff(; compile = true)
+    chn = sample(mdl, NUTS(adtype = ad), MCMCThreads(), 1000, 4, progess = false)
+
+    @testset "Check model vals and grads agree across AD systems for random param choice" begin
+        ℓ = DynamicPPL.LogDensityFunction(mdl)
+        DynamicPPL.link!!(ℓ.varinfo, mdl)
+
+        n = LogDensityProblems.dimension(ℓ)
+        LogDensityProblems.logdensity(ℓ, zeros(n))
+        ∇ℓ_rd = LogDensityProblemsAD.ADgradient(Val(:ReverseDiff), ℓ)
+        ∇ℓ_rd2 = ADgradient(:ReverseDiff, ℓ; compile = Val(true))
+        ∇ℓ_fd = LogDensityProblemsAD.ADgradient(Val(:ForwardDiff), ℓ)
+
+        x = randn(n)
+        val1, g1 = LogDensityProblems.logdensity_and_gradient(∇ℓ_rd, x)
+        val2, g2 = LogDensityProblems.logdensity_and_gradient(∇ℓ_fd, x)
+        val3, g3 = LogDensityProblems.logdensity_and_gradient(∇ℓ_rd2, x)
+
+        @test val1 ≈ val2 ≈ val3
```

Comment:
do we need this given we have benchmarking also doing this? Thoughts?

Reply:
Ah, this was my idiot checking that got a bit carried away. This should just replicate the Turing benchmarks so can get removed. 

Maybe is tests the `Turing.Model` -> `LogDensityProblem` conversion but... I think this is just exactly what Turing benchmarks does anyway.